### PR TITLE
Add backtesting menu.

### PR DESCRIPTION
**Pull request by jmaslek**
This PR adds a couple super simple strategies to test ideas on.  I added buy and hold comparisons along with spy return comparison.

For some reason, the bt library results in a change of the matplotlib backend, so to address that, I added another line to config_plot that allows the user to specify which they want to use to prevent the "agg" from being defaulted to.

I also raise an error if a non-gun backend is used.

### CODE REVIEW

**Review by aia — APPROVED** [review]

**Comment by DidierRLopes** [comment]
I'm getting this @jmaslek :

```
(✨) (bt)> ema 
/Users/DidierRodriguesLopes/Documents/git/GamestonkTerminal/gamestonk_terminal/backtesting/bt_view.py:79: UserWarning: Matplotlib is currently using agg, which is a non-GUI backend, so cannot show the figure.
  plt.show()
```

File: gamestonk_terminal/backtesting/bt_controller.py [hunk 1]
```diff
@@ -0,0 +1,114 @@
...
+    bt_controller.call_help(None)
+
+    while True:
+        # Get input command from user
+        if session and gtff.USE_PROMPT_TOOLKIT:
+            completer = NestedCompleter.from_nested_dict(
+                {c: None for c in bt_controller.CHOICES}
+            )
+            an_input = session.prompt(
+                f"{get_flair()} (ta)> ",
```
Comment (DidierRLopes) [inline gamestonk_terminal/backtesting/bt_controller.py:98]: Should be (bt)

File: gamestonk_terminal/backtesting/bt_view.py [hunk 2]
```diff
@@ -0,0 +1,288 @@
...
+                stock_bt = buy_and_hold(ticker, start_date, ticker.upper() + " Hold")
+                res = bt.run(bt_backtest, spy_bt, stock_bt)
+        else:
+            if ns_parser.no_bench:
+                res = bt.run(bt_backtest)
+            else:
+                stock_bt = buy_and_hold(ticker, start_date, ticker.upper() + " Hold")
+                res = bt.run(bt_backtest, stock_bt)
+
+        _, ax = plt.subplots(figsize=plot_autoscale(), dpi=PLOT_DPI)
```
Comment (DidierRLopes) [inline gamestonk_terminal/backtesting/bt_view.py:71]: probably have this data on a `plot_backtest` function to follow MVC design.

Same for others

File: gamestonk_terminal/backtesting/README.md [hunk 3]
```diff
@@ -0,0 +1,59 @@
+# Backtesting
```
Comment (DidierRLopes) [inline gamestonk_terminal/backtesting/README.md:1]: Don't worry with README. I'll take care of this :)

**Comment by jmaslek** [comment]
I added the pandas register ppl converters so hopefully that allows you to plot?

I have no issues

**Comment by DidierRLopes** [comment]
> I added the pandas register ppl converters so hopefully that allows you to plot?
> 
> I have no issues

let me test this now!

File: gamestonk_terminal/backtesting/bt_controller.py [hunk 4]
```diff
@@ -0,0 +1,116 @@
...
+    def call_q(self, _):
+        """Process Q command - quit the menu"""
+        return False
+
+    def call_quit(self, _):
+        """Process Quit command - quit the program"""
+        return True
+
+    def call_ema(self, other_args: List[str]):
+        bt_view.simple_ema(self.ticker, self.start, other_args)
```
Comment (DidierRLopes) [inline gamestonk_terminal/backtesting/bt_controller.py:76]: You forgot the comment here and in the 2 below as well

File: gamestonk_terminal/backtesting/bt_helper.py [hunk 5]
```diff
@@ -0,0 +1,18 @@
+from datetime import datetime
+from typing import Union
+import bt
+
+
+def buy_and_hold(ticker: str, start: Union[str, datetime], name: str):
```
Comment (DidierRLopes) [inline gamestonk_terminal/backtesting/bt_helper.py]: docstring missing here

File: gamestonk_terminal/backtesting/bt_helper.py [hunk 6]
```diff
@@ -0,0 +1,51 @@
...
+from pandas.plotting import register_matplotlib_converters
+import bt
+from gamestonk_terminal.helper_funcs import plot_autoscale
+from gamestonk_terminal.config_plot import PLOT_DPI
+from gamestonk_terminal import feature_flags as gtff
+
+register_matplotlib_converters()
+
+
+def buy_and_hold(ticker: str, start: Union[str, datetime], name: str):
```
Comment (DidierRLopes) [inline gamestonk_terminal/backtesting/bt_helper.py:17]: docstring missing here

File: gamestonk_terminal/backtesting/bt_view.py [hunk 7]
```diff
@@ -0,0 +1,255 @@
...
+from typing import List, Union
+from datetime import datetime
+import pandas as pd
+import bt
+import pandas_ta as ta
+from gamestonk_terminal.helper_funcs import parse_known_args_and_warn
+from gamestonk_terminal.backtesting.bt_helper import buy_and_hold, plot_bt
+
+
+def simple_ema(ticker: str, start_date: Union[str, datetime], other_args: List[str]):
```
Comment (DidierRLopes) [inline gamestonk_terminal/backtesting/bt_view.py:13]: docstring missing here

File: gamestonk_terminal/backtesting/bt_view.py [hunk 8]
```diff
@@ -0,0 +1,255 @@
...
+        print(res.display())
+        print("")
+
+    except Exception as e:
+        print(e)
+        print("")
+        return
+
+
+def ema_cross(ticker: str, start_date: Union[str, datetime], other_args: List[str]):
```
Comment (DidierRLopes) [inline gamestonk_terminal/backtesting/bt_view.py:79]: docstring missing here

File: gamestonk_terminal/backtesting/bt_view.py [hunk 9]
```diff
@@ -0,0 +1,255 @@
...
+        print(res.display())
+        print("")
+
+    except Exception as e:
+        print(e)
+        print("")
+        return
+
+
+def rsi_strat(ticker: str, start_date: Union[datetime, str], other_args: List[str]):
```
Comment (DidierRLopes) [inline gamestonk_terminal/backtesting/bt_view.py:166]: docstring missing here

**Comment by jmaslek** [comment]
I added to config_plot the ability to set the MPL backend.  I defaulted it to None since there are no issues.  Once this is good to merge, we may need to add troubleshooting?

File: gamestonk_terminal/config_plot.py [hunk 10]
```diff
@@ -1,5 +1,19 @@
...
+More information:
+https://matplotlib.org/stable/tutorials/introductory/usage.html#the-builtin-backends
+
+Common ones:
+module://backend_interagg   : This is what pycharm defaults to in Scientific Mode
+MacOSX   : Mac default.  Does not work with backtesting
+tkAgg : This uses the tkinter library.
+Qt5Agg  : This requires the PyQt5 package is installed
+"""
+backend = None
```
Comment (DidierRLopes) [inline gamestonk_terminal/config_plot.py:15]: Maybe add this in description of PR " After using the bt library, which somehow changes the
# backend to 'agg' it raises an error.  

Could we re-write this as, and put it at the bottom? 

```
# Backend to use for plotting
backend = None
# Examples:
# "module://backend_interagg" - This is what pycharm defaults to in Scientific Mode
# "MacOSX" - Mac default.  Does not work with backtesting
# "tkAgg" - This uses the tkinter library.
# "Qt5Agg" - This requires the PyQt5 package is installed
# See more: https://matplotlib.org/stable/tutorials/introductory/usage.html#the-builtin-backends
```

Comment (jmaslek) [reply gamestonk_terminal/config_plot.py:15]: I can make it like that (should I comment each line or just keep the doctoring format?)

And I will update the PR description now.

Comment (DidierRLopes) [reply gamestonk_terminal/config_plot.py]: As you have it is good! Once you fix the linter, feel free to merge this!

**Pull request closed without merging**